cmd/kube-proxy/app/init_windows.go

```diff
@@ -0,0 +1,44 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"k8s.io/kubernetes/pkg/windows/service"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	serviceName = "kube-proxy"
+)
+
+var (
+	flRunService bool
```

Comment:
@ncdc this global state look good? Or does kube-proxy prefer something akin to `KubeletFlags`?

Reply:
I had put everything into the `Options` struct, such as `CleanupAndExit`:

https://github.com/kubernetes/kubernetes/blob/86ca3afb4d03c769d4e643eb0320f17cdf7f9903/cmd/kube-proxy/app/server.go#L93-L100

And then the flag registered directly into that field:

https://github.com/kubernetes/kubernetes/blob/86ca3afb4d03c769d4e643eb0320f17cdf7f9903/cmd/kube-proxy/app/server.go#L120-L124

But I realize this is OS specific and it doesn't make sense to add it to kube-proxy's Options struct, so it's probably ok like this.

FWIW, I wouldn't prefix the package variable's name with `fl`. I'd probably call it `runAsWIndowsServer` or something to that effect instead.

Reply:
Done. I changed the variable name into ``WindowsService`` to be consistent with Kubelet as @mtaufen requested.